react/features/filmstrip/middleware.web.js

```diff
@@ -240,6 +248,36 @@ MiddlewareRegistry.register(store => next => action => {
         }
         break;
     }
+    case TOGGLE_PIN_STAGE_PARTICIPANT: {
+        const { dispatch, getState } = store;
+        const state = getState();
+        const { participantId } = action;
+        const pinnedParticipants = getPinnedActiveParticipants(state);
+        const dominant = getDominantSpeakerParticipant(state);
+
+        if (pinnedParticipants.find(p => p.participantId === participantId)) {
+            if (dominant?.id === participantId) {
+                const { activeParticipants } = state['features/filmstrip'];
+                const queue = activeParticipants.map(p => {
```

Comment:
so this ensures dominant speaker is no longer pinned as soon as he's done being the dominant speaker right?